libs/portal-plugin-ipfs/src/ui/components/toolbar/PinItem.tsx

```
@@ -0,0 +1,62 @@
+
+export function PinItem<
+  TData extends BaseRecord = any,
+>(): ToolbarCustomItem<TData> {
+  const { openDialog } = useDialog();
+  const { open } = useNotification();
+
+  return {
+    id: "add-pin",
+    type: ToolbarItemType.CUSTOM,
+    component: () => {
+      const { getHeliaService } = useFileManagerFeature();
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)